core/src/main/java/org/geysermc/geyser/GeyserImpl.java

```diff
@@ -154,6 +155,8 @@ private GeyserImpl(PlatformType platformType, GeyserBootstrap bootstrap) {
         MessageTranslator.init();
         MinecraftLocale.init();
 
+        GeyserExtensionManager.init();
```

Comment:
If extensions are not enabled in start() then a reload caused by the server will permanently disable extensions, afaik. 

I would remove the enabling of extensions from extensionmanager#init() and then enable them all in GeyserImpl#start()